File: samples/dark_castle.r
# In a land shrouded in mystery...

castle_name <- "Dark Castle"
construction_year <- 1100
destruction_year <- 1678

calc_duration <- function(construction_year, destruction_year) {
  return(destruction_year - construction_year)
}

duration <- calc_duration(construction_year, destruction_year)
cat("The duration of", castle_name, "was", duration, "years.\n")

# ...there stood a Dark Castle

File: samples/dark_castle.lua
-- In a land shrouded in mystery...

local castleName = "Dark Castle"
local constructionYear = 1100
local destructionYear = 1678

local function calcDuration()
  return destructionYear - constructionYear
end

print("The duration of " .. castleName .. " was " .. calcDuration() .. " years.")

-- ...there stood a Dark Castle
